### PR TITLE
9.0 migration of mail_environment

### PR DESCRIPTION
**Pull request by gurneyalex**

### CODE REVIEW

**Comment by elicoidal** [comment]
shall we comment or is it too soon?

**Comment by gurneyalex** [comment]
depends on #337

**Comment by gurneyalex** [comment]
@elicoidal reviews welcome

File: mail_environment/README.rst [hunk 1]
```diff
@@ -0,0 +1,113 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+   :target: http://www.gnu.org/licenses/agpl-3.0-standalone.html
+   :alt: License: AGPL-3
+
+================
+mail environment
```
Comment (elicoidal) [inline mail_environment/README.rst]: Environment File for Mail Configuration

File: mail_environment/README.rst [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+================
+
+This module allows to configure the incoming and outgoing mail servers
+using the server_environment mechanism.
```
Comment (elicoidal) [inline mail_environment/README.rst]: in a separate configuration file using server_environment mechanism.

File: mail_environment/README.rst [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+============
+
+To install this module, you need to have the server_environment module
+installed and properly configured.
+
+Configuration
+=============
+
+With this module installed, the incoming and outgoing mail servers are
+in the server_environment_files module that you should provide.
```
Comment (elicoidal) [inline mail_environment/README.rst]: set up in a file that is referenced in server_environment_files module.

File: mail_environment/README.rst [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+
+In the configuration file for each environment, you may use the
```
Comment (elicoidal) [inline mail_environment/README.rst]: In the configuration file of each environment, you may first use the
sections 'outgoing_mail' and 'incoming_mail' to configure the default
values respectively for SMTP servers and the IMAP/POP servers.

File: mail_environment/README.rst [hunk 5]
```diff
@@ -0,0 +1,113 @@
...
+sections 'outgoing_mail' and 'incoming_mail' to configure the default
+values for SMTP servers and the IMAP/POP servers respectively.
+
+Then for each server, you can (re)define values with a section named
```
Comment (elicoidal) [inline mail_environment/README.rst]: Then for each server, you can (re)define additional values with a section named

File: mail_environment/README.rst [hunk 6]
```diff
@@ -0,0 +1,113 @@
...
+"outgoing_mail.resource_name" or "incoming_mail.resource_name" where
+"resource_name" is the name of your server.
+
+Exemple of config file :
```
Comment (elicoidal) [inline mail_environment/README.rst:36]: remove space before colon

File: mail_environment/README.rst [hunk 7]
```diff
@@ -0,0 +1,113 @@
...
+
+[outgoing_mail]
+smtp_host = smtp.myserver.com
+smtp_port = 587
+smtp_user =
+smtp_pass =
+smtp_encryption = ssl
+
+[outgoing_mail.openerp_smtp_server1]
+smtp_user = openerp
```
Comment (elicoidal) [inline mail_environment/README.rst]: s/openerp/odoo (x4)

File: mail_environment/README.rst [hunk 8]
```diff
@@ -0,0 +1,113 @@
...
+attach = 0
+original = 0
+user = openerp@myserver.com
+password = openerp
+
+
+Usage
+=====
+
+Once configured, your the values for the mail servers in the Odoo
```
Comment (elicoidal) [inline mail_environment/README.rst]: Once configured, Odoo will read the mail servers values from the 
configuration file related to each environment defined in the main Odoo file.

File: mail_environment/__openerp__.py [hunk 9]
```diff
@@ -21,44 +21,12 @@
 
 {
     'name': 'Server env config for mail + fetchmail',
```
Comment (elicoidal) [inline mail_environment/__openerp__.py]: Environment File for Mail Configuration

File: mail_environment/__openerp__.py [hunk 10]
```diff
@@ -21,44 +21,12 @@
```
Comment (elicoidal) [inline mail_environment/__openerp__.py:18]: Please use short headers

File: mail_environment/__openerp__.py [hunk 11]
```diff
@@ -21,44 +21,12 @@
...
-type = pop
-is_ssl = 0
-attach = 0
-original = 0
-user = openerp@myserver.com
-password = openerp
-
-
-    """,
+    'summary': 'configure mail servers with server_environment_files',
```
Comment (elicoidal) [inline mail_environment/__openerp__.py]: s/configure/Configure

File: mail_environment/__openerp__.py [hunk 12]
```diff
@@ -21,44 +21,12 @@
...
     'author': "Camptocamp,Odoo Community Association (OCA)",
     'license': 'AGPL-3',
-    'website': 'http://openerp.camptocamp.com',
+    'website': 'http://odoo-community.org',
```
Comment (elicoidal) [inline mail_environment/__openerp__.py:26]: I think you should keep the main website of c2c

File: mail_environment/__openerp__.py [hunk 13]
```diff
@@ -21,44 +21,12 @@
...
     'depends': ['mail',
                 'fetchmail',
```
Comment (elicoidal) [inline mail_environment/__openerp__.py]: Is it necessary? fetchmail depends on mail already

File: mail_environment/__openerp__.py [hunk 14]
```diff
@@ -66,6 +34,6 @@
                 ],
     'data': ['mail_view.xml'],
     'demo': [],
```
Comment (elicoidal) [inline mail_environment/__openerp__.py]: remove empty keys

File: mail_environment/README.rst [hunk 15]
```diff
@@ -0,0 +1,113 @@
...
+[incoming_mail.openerp_pop_mail1]
+server = mail.myserver.com
+port = 110
+type = pop
+is_ssl = 0
+attach = 0
+original = 0
+user = openerp@myserver.com
+password = openerp
+
```
Comment (elicoidal) [inline mail_environment/README.rst:62]: Example is great. Would it be useful to add screenshots?

File: mail_environment/__openerp__.py (hunk 13 above)
Comment (lmignon) [inline mail_environment/__openerp__.py]: @elicoidal IMO a module must declare all these direct dependencies even if one of them is transitively achieved. In this case, even if mail no more depdends of fetchmail, the dependencies will be honored.

Comment (elicoidal) [inline mail_environment/__openerp__.py]: @lmignon I am not against and not sure whether there is a guideline for that

File: mail_environment/README.rst (hunk 15 above)
Comment (gurneyalex) [inline mail_environment/README.rst:62]: screenshots of what?

File: mail_environment/__openerp__.py (hunk 13 above)
Comment (sebalix) [inline mail_environment/__openerp__.py]: I agree with @lmignon to be explicit regarding direct dependencies of a module. It allows to understand what this module extends (through XML-ID, data models and methods...), and reduces the scope of research of original resources.

**Comment by elicoidal** [comment]
LGTM

File: mail_environment/README.rst [hunk 16]
```diff
@@ -0,0 +1,121 @@
...
+To install this module, you need to have the server_environment module
+installed and properly configured.
+
+Configuration
+=============
+
+With this module installed, the incoming and outgoing mail servers are
+configured in the `server_environment_files` module (which is a module
+you should provide, see the documentaion of `server_environment` for
```
Comment (guewen) [inline mail_environment/README.rst]: s/documentaion/documentation/

**Comment by guewen** [comment]
:+1:

**Comment by mdietrichc2c** [comment]
:+1:

**Comment by leemannd** [comment]
:+1:

**Pull request merged**